full-stack-tests/presentation/src/frontend/UnifiedSelection.test.ts

```diff
@@ -185,6 +188,172 @@ describe("Unified Selection", () => {
+        };
+        return builder.insertElement(physicalObjectProps);
+      };
+
+      const objectIdArray = Array<Id64String>();
+      let groupingNodesIModel: IModelConnection;
+      beforeEach(async () => {
+        groupingNodesIModel = await buildTestIModel("GroupByClass", (builder) => {
+          const physicalPartitionId = addPartition("BisCore:PhysicalPartition", builder, "TestDrawingModel");
+          const definitionPartitionId = addPartition("BisCore:DefinitionPartition", builder, "TestDefinitionModel");
+          const physicalModelId = addModel("BisCore:PhysicalModel", builder, physicalPartitionId);
+          const definitionModelId = addModel("BisCore:DefinitionModel", builder, definitionPartitionId);
+          const categoryId = addSpatialCategory(builder, definitionModelId, "Test SpatialCategory");
+          objectIdArray.push(addPhysicalObject(builder, physicalModelId, categoryId, new Code({ value: "code", scope: "0x1", spec: "0x1" })));
+          objectIdArray.push(addPhysicalObject(builder, physicalModelId, categoryId, new Code({ value: "code", scope: "0x1", spec: "0x2" })));
+          objectIdArray.push(addPhysicalObject(builder, physicalModelId, categoryId, new Code({ value: "code2", scope: "0x1", spec: "0x2" })));
+        });
+
+        Presentation.selection.clearSelection("", groupingNodesIModel);
+        handler = new ViewportSelectionHandler({ imodel: groupingNodesIModel });
+      });
+
+      afterEach(async () => {
+        await Presentation.presentation.rulesets().clear();
+        objectIdArray.splice(0);
+        await groupingNodesIModel.close();
+        handler.dispose();
+      });
+
+      it("hilites grouping nodes grouped by class", async () => {
+        const groupByClassRuleset: Ruleset = {
+          id: "groupByClassRuleset",
+          rules: [{
+            ruleType: RuleTypes.RootNodes,
+            specifications: [{
+              specType: ChildNodeSpecificationTypes.InstanceNodesOfSpecificClasses,
+              classes: { schemaName: "BisCore", classNames: [GeometricElement3d.className] },
+              arePolymorphic: true,
+              instanceFilter: "this.Parent = NULL",
+              groupByClass: true,
+              groupByLabel: false,
+            }],
+          }],
+        };
+
+        await using<RegisteredRuleset, Promise<void>>(await Presentation.presentation.rulesets().add(groupByClassRuleset), async () => {
+          const rootNodes = await Presentation.presentation.getNodes({ imodel: groupingNodesIModel, rulesetOrId: groupByClassRuleset.id });
+          Presentation.selection.addToSelection("", groupingNodesIModel, [rootNodes[0].key]);
+          await waitForAllAsyncs([handler, Presentation.selection.getToolSelectionSyncHandler(groupingNodesIModel)!]);
+          expect(groupingNodesIModel.hilited.elements.size).to.be.equal(3);
+          objectIdArray.forEach((id) => expect(groupingNodesIModel.hilited.elements.hasId(id)).to.be.true);
```

Comment:
please rename `objectIdArray` to something more meaningful. "object" is way too generic, it's not clear what objects you have in mind.

Reply:
Renamed to `physicalObjectIdArray`. Any better suggestions?